### DISCUSSION TITLE
Allow regex / non fuzzy search in the file picker

### DISCUSSION THREAD

**Discussion by yoav-lavi** [helix-editor/helix#8181] [Ideas] [answered]
For certain workflows (e.g. "go over every `.rs` file in directory `xyz`") it would be beneficial to be able to more strictly filter files in the file picker.

This could be done by defining a specific syntax for it (e.g. `/query/` is regex, `!query` is exact) or using some sort of keybinding to switch modes with a UI indication of the current mode.

**Comment by yoav-lavi** [comment]
An additional mode or syntax could be search in files (like `space`+`/`, but in a file selection context)

**Comment by pascalkuthe** [comment, marked as answer]
Fuzzy completions already have special syntax for more exact searches.

You can use `$` for suffix match, `^` for prefix match, `'` for substring match, `!` for negated substring match (every that contains this substring will be removed), `!^` for negated prefix match and `!...$` for negated postifx match.


This syntax allows what you have described. For example

`^directory/to/search .rs$`


I don't see us implementing a regex mode for the pickers.

**Reply by yoav-lavi** [reply]
Thank you!
